enhancements/installer/storage-class-encrypted.md

```diff
@@ -0,0 +1,350 @@
+  - "@joelspeed"
+creation-date: 2022-06-22
+last-updated: 2022-12-09
+tracking-link:
+  - https://issues.redhat.com/browse/CORS-2049
+  - https://issues.redhat.com/browse/OCPPLAN-9307
+---
+
+# Default Storage Class with Encrypted Keys 
+
+## Summary
+
+When user-managed encryption keys are provided at install time, the default storage
+class should be set to utilize those keys. Presently, only root volumes for nodes are
+encrypted with those keys--but users would expect the persistent storage volumes to
+be encrypted as well. This enhancement proposes a solution for AWS, GCP, & Azure.
+
+## Motivation
+
+The primary motivation is to enable managed services to encrypt storage volumes with
+customer-provided keys automatically as part of the installation process without requiring
+day-2 operations. The behavior of automatically setting the default storage class is
+sensible for individual (non-managed-service) users as well; setting the default will
+reduce their day-2 burden and decrease the opportunity for user error.
+
+### User Stories
+
+When I install a cluster, I want to provide user-managed encryption keys in the install-config
+so that the default storage class will be set to provision volumes with those keys.
+
+### Goals
+
+* Default storage class is set to use user-provided keys at cluster creation for public clouds (AWS, GCP, Azure)
+
+### Non-Goals
+* Other clouds besides AWS, GCP, & Azure are out of scope
+
+
+## Proposal
+
+### Workflow Description
+
+1. Cluster creator specifies (aws|gcp|azure) user-managed encryption keys 
+in `defaultMachinePlatform` of install config.
+2. Installer adds encryption key fields to cluster infrastructure object.
+3. Cluster-storage-operator/CSI drivers use cluster infrastructure object
+to create default storage class using user-encrypted keys from infra object.
+
+
+### API Extensions
+
+This enhancement extends the cluster `Infrastructure` API object to add fields
+for the user-managed encryption keys. The fields will be added to the platform
+spec and status for each platform. They are added to the spec so that users can edit the
+spec day 2 in order to update the default storage class. The status field would
+be updated by the cluster-storage operator/CSI driver.
+
+#### AWS 
+
+```go
+// AWSPlatformSpec holds the desired state of the Amazon Web Services infrastructure provider.
+// This only includes fields that can be modified in the cluster.
+type AWSPlatformSpec struct {
+	// [... existing fields ...]
+
+    // The KMS key to be used in the default storage class to encrypt volumes.
+    // +optional
+    KMSKeyARN string `json:"kmsKeyARN,omitempty"`
+}
+
+// AWSPlatformStatus holds the current status of the Amazon Web Services infrastructure provider.
+type AWSPlatformStatus struct {
+    // [... existing fields ...]
+
+    // The KMS key used by the default storage class to encrypt volumes.
+    // +optional
+    KMSKeyARN string `json:"kmsKeyARN,omitempty"`
```

Comment:
Why do we have this in spec and status? What's the difference? This will need to be explained within the godoc

Reply:
https://github.com/openshift/enhancements/pull/1163#discussion_r1046083045